gpkit/small_classes.py

```diff
@@ -137,16 +136,21 @@ def index_dict(idx, i, o):
     return o
 
 
-def enray_dict(i, o):
+def enray_and_unit_dict(i, o, unitless_keys=[], united=False):
     "Recursively turns lists into numpy arrays."
     for k, v in i.items():
         if isinstance(v, dict):
-            o[k] = enray_dict(v, {})
+            if k in unitless_keys:
+                united = False
+            o[k] = enray_and_unit_dict(v, {}, unitless_keys, united)
         else:
             if len(v) == 1:
-                o[k] = np.array(v[0])
-            else:
-                o[k] = np.array(v)
+                v = v[0]
+            v = np.array(v)
+            if (united and hasattr(k, "units")
+                    and isinstance(k.units, Quantity)):
+                v = v*k.units
+            o[k] = v
     # assert set(i.keys()) == set(o.keys())  # keys change with swept varkeys
     return o
```

Comment:
I see, it's really a help method for `DictOfLists`. Should it be private (i.e. `_enray_and_unit_dict`)?